iothub/device/src/ConnectionInfo.cs

```diff
@@ -15,13 +15,15 @@ internal ConnectionInfo()
             Status = ConnectionStatus.Disconnected;
             ChangeReason = ConnectionStatusChangeReason.ClientClose;
             StatusLastChangedOnUtc = DateTimeOffset.UtcNow;
+            RecommendedAction = RecommendedAction.DefaultAction;
         }
 
-        internal ConnectionInfo(ConnectionStatus status, ConnectionStatusChangeReason changeReason, DateTimeOffset changedOnUtc)
-        { 
+        internal ConnectionInfo(ConnectionStatus status, ConnectionStatusChangeReason changeReason)
+        {
             Status = status;
             ChangeReason = changeReason;
-            StatusLastChangedOnUtc = changedOnUtc;
+            StatusLastChangedOnUtc = DateTimeOffset.UtcNow;
+            RecommendedAction = GetRecommendedActionBasedOnConnectionStatusAndChangeReason(status, changeReason);
```

Comment:
Long and clear method..... but what about a shorter one?
GetRecommendedAction(connectionStatus, changeReason)